--- a/kubernetes/dockerfiles/spark/bindings/python/Dockerfile
+++ b/kubernetes/dockerfiles/spark/bindings/python/Dockerfile
@@ -32,6 +32,8 @@ RUN apk add --no-cache python && \
     # Removed the .cache to save space
     rm -r /root/.cache

+# Backport SPARK-26083 to get working pyspark shell
+COPY python/pyspark ${SPARK_HOME}/python/pyspark
 COPY python/lib ${SPARK_HOME}/python/lib
 ENV PYTHONPATH ${SPARK_HOME}/python/lib/pyspark.zip:${SPARK_HOME}/python/lib/py4j-*.zip
