--- a/kubernetes/dockerfiles/spark/Dockerfile
+++ b/kubernetes/dockerfiles/spark/Dockerfile
@@ -48,6 +48,12 @@ COPY examples /opt/spark/examples
 COPY ${k8s_tests} /opt/spark/tests
 COPY data /opt/spark/data

+# Install hadoop-aws to the image
+RUN java -jar /opt/spark/jars/ivy-2.4.0.jar \
+         -dependency org.apache.hadoop hadoop-aws 3.2.0 \
+         -retrieve '/opt/spark/jars/[organization]_[artifact]-[revision](-[classifier]).[ext]' && \
+    rm -rf /root/.ivy2/
+
 ENV SPARK_HOME /opt/spark

 WORKDIR /opt/spark/work-dir
